src/libData/BlockData/Block/MicroBlock.cpp

```diff
@@ -26,14 +26,11 @@ using namespace boost::multiprecision;
 unsigned int MicroBlock::Serialize(vector<unsigned char>& dst,
                                    unsigned int offset) const
 {
-    if (m_header.GetNumTxs() != m_tranHashes.size())
-    {
-        LOG_GENERAL(WARNING,
-                    "assertion failed (" << __FILE__ << ":" << __LINE__ << ": "
-                                         << __FUNCTION__ << ")");
-    }
+    unsigned int size_core = SerializeCore(dst, offset);
+
+    offset += size_core;
 
-    unsigned int size_needed = GetSerializedSize();
+    unsigned int size_needed = GetSerializedTxnHashesSize();
```

Comment:
Even though the tx hashes are no longer added to the serialized message, it looks like we're still resizing the destination buffer to include the size of the hashes, and then also iterating through the hashes without actually doing anything to them.  Perhaps we can get rid of this part already?